packages/rum-core/src/boot/preStartRum.ts

```diff
@@ -0,0 +1,168 @@
+  noop,
+  timeStampNow,
+  clocksNow,
+  assign,
+} from '@datadog/browser-core'
+import {
+  validateAndBuildRumConfiguration,
+  type RumConfiguration,
+  type RumInitConfiguration,
+} from '../domain/configuration'
+import type { CommonContext } from '../domain/contexts/commonContext'
+import type { ViewOptions } from '../domain/view/trackViews'
+import type { RumPublicApiOptions, Strategy } from './rumPublicApi'
+import type { StartRumResult } from './startRum'
+
+export function createPreStartStrategy(
+  { ignoreInitIfSyntheticsWillInjectRum, startDeflateWorker }: RumPublicApiOptions,
+  getCommonContext: () => CommonContext,
+  doStartRum: (
+    initConfiguration: RumInitConfiguration,
+    configuration: RumConfiguration,
+    deflateWorker: DeflateWorker | undefined,
+    initialViewOptions?: ViewOptions
+  ) => StartRumResult
+): Strategy {
+  const bufferApiCalls = new BoundedBuffer<StartRumResult>()
+  let firstStartViewCall:
+    | { options: ViewOptions | undefined; callback: (startRumResult: StartRumResult) => void }
+    | undefined
+  let deflateWorker: DeflateWorker | undefined
+
+  let cachedInitConfiguration: RumInitConfiguration | undefined
+  let cachedConfiguration: RumConfiguration | undefined
+
+  function tryStartRum() {
+    if (!cachedInitConfiguration || !cachedConfiguration) {
+      return
+    }
+
+    let initialViewOptions: ViewOptions | undefined
+
+    if (cachedConfiguration.trackViewsManually) {
+      if (!firstStartViewCall) {
+        return
+      }
+      // An initial view is always created when starting RUM.
+      // When tracking views automatically, any startView call before RUM start creates an extra
+      // view.
+      // When tracking views manually, we use the ViewOptions from the first startView call as the
+      // initial view options, and we remove the actual startView call so we don't create
+      bufferApiCalls.remove(firstStartViewCall.callback)
+      initialViewOptions = firstStartViewCall.options
+    }
+
+    const startRumResult = doStartRum(cachedInitConfiguration, cachedConfiguration, deflateWorker, initialViewOptions)
+
+    bufferApiCalls.drain(startRumResult)
+  }
+
+  return {
+    init(initConfiguration) {
+      if (!initConfiguration) {
+        display.error('Missing configuration')
+        return
+      }
+
+      const eventBridgeAvailable = canUseEventBridge()
+      if (eventBridgeAvailable) {
+        initConfiguration = overrideInitConfigurationForBridge(initConfiguration)
+      }
+
+      // Expose the initial configuration regardless of initialization success.
+      cachedInitConfiguration = initConfiguration
+
+      if (cachedConfiguration) {
+        displayAlreadyInitializedError('DD_RUM', initConfiguration)
+        return
+      }
+
+      // If we are in a Synthetics test configured to automatically inject a RUM instance, we want to
+      // completely discard the customer application RUM instance by ignoring their init() call.  But,
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏extra space
```suggestion
      // completely discard the customer application RUM instance by ignoring their init() call. But,
```